### PR TITLE
DAO-385: Implement Upgradable veRIF ERC20 contract with ERC20 Votes

### PR DESCRIPTION
**Pull request by rodrigoncalves**
The unit tests will go in separate ticket.

My guess is we'll end up with "veRIF". We can change this if the product folks thinks so. 

I took the liberty of configuring prettier in this PR.

### CODE REVIEW

**Review by Freshenext — APPROVED** [review]
Looks good!

File: hardhat.config.ts [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
+import '@nomicfoundation/hardhat-toolbox'
+import '@openzeppelin/hardhat-upgrades'
 
 const config: HardhatUserConfig = {
   solidity: {
-    compilers: [
-      { version: '0.8.24' }, 
-      { version: '0.4.24' }
-    ]
+    compilers: [{ version: '0.8.24' }, { version: '0.4.24' }],
```
Comment (TravellerOnTheRun) [inline hardhat.config.ts:7]: Quick question: why is 0.4.24 here?

**Review by TravellerOnTheRun — COMMENTED** [review]
Is that all regarding tests? Or the rest is out of scope?

**Review by sleyter93 — DISMISSED** [review]
Implementation looks good. We should include some test for veRIFToken. I believe the wrapper and voting functions

File: hardhat.config.ts (hunk 1 above)
Comment (rodrigoncalves) [reply hardhat.config.ts:7]: Because RIF token contract.

**Comment by rodrigoncalves** [comment]
> Implementation looks good. We should include some test for veRIFToken. I believe the wrapper and voting functions

The unit tests will go in another ticket.

**Pull request merged**